tests/wp-unit/include/Core/Pages/Metabox/DeletePagesByStatusMetaboxTest.php

```diff
@@ -75,12 +98,142 @@ public function test_that_published_pages_can_be_deleted() {
 		);
 
 		$pages_deleted = $this->metabox->delete( $delete_options );
-		$this->assertEquals( 10, $pages_deleted );
+		$this->assertEquals( 20, $pages_deleted );
+
+		$published_pages = $this->get_pages_by_status();
+		$this->assertEquals( 0, count( $published_pages ) );
+
+		$trash_pages = $this->get_pages_by_status( 'trash' );
+		$this->assertEquals( 0, count( $trash_pages ) );
+	}
+
+	/**
+	 * Test date filter (older than x days) with a single post status.
+	 */
+	public function test_that_pages_that_are_older_than_x_days_can_be_deleted() {
+		$date = date( 'Y-m-d H:i:s', strtotime( '-5 day' ) );
+
+		$published_pages = $this->factory->post->create_many( 10, array(
+			'post_type'   => 'page',
+			'post_status' => 'publish',
+			'post_date'   => $date,
+		) );
+
+		$this->assertEquals( 10, count( $published_pages ) );
+
+		$draft_pages = $this->factory->post->create_many( 10, array(
+			'post_type'   => 'page',
+			'post_status' => 'draft',
+			'post_date'   => $date,
+		) );
+
+		$this->assertEquals( 10, count( $draft_pages ) );
+
+		$delete_options = array(
+			'publish'      => 'published_pages',
+			'drafts'       => 'draft_pages',
+			'pending'      => '',
+			'future'       => '',
+			'private'      => '',
+			'limit_to'     => -1,
+			'restrict'     => true,
+			'force_delete' => false,
+			'date_op'      => 'before',
+			'days'         => '3',
+		);
+
+		$pages_deleted = $this->metabox->delete( $delete_options );
+		$this->assertEquals( 20, $pages_deleted );
+
+		$published_pages = $this->get_pages_by_status();
+		$this->assertEquals( 0, count( $published_pages ) );
+
+		$trash_pages = $this->get_pages_by_status( 'trash' );
+		$this->assertEquals( 20, count( $trash_pages ) );
+	}
+
+	/**
+	 * Test date filter (within the last x days) with a single post status.
```

Comment:
The comment says single post status, but inside you are creating two post status.